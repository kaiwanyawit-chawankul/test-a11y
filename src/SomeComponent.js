import React from 'react';

const SomeComponent = () => {
    return (
        <div role="main">
            <button>Click me</button>
        </div>
    );
};

export default SomeComponent;