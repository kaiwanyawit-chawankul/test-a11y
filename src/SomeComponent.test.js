import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { axe, toHaveNoViolations } from 'jest-axe';
import userEvent from '@testing-library/user-event';
import SomeComponent from './SomeComponent';

expect.extend(toHaveNoViolations);

describe('SomeComponent', () => {
    it('renders without crashing', () => {
        render(<SomeComponent />);
    });

    it('displays correct content', () => {
        render(<SomeComponent />);
        expect(screen.getByRole('main')).toBeInTheDocument();
    });

    it('meets accessibility requirements', async () => {
        const { container } = render(<SomeComponent />);
        const results = await axe(container, {
            rules: {
                'aria-roles': { enabled: true },
                'button-name': { enabled: true },
                'image-alt': { enabled: true }
            }
        });
        expect(results).toHaveNoViolations();
    });

    it('handles user interactions', async () => {
        const user = userEvent.setup();
        render(<SomeComponent />);
        const button = screen.getByRole('button');
        await user.click(button);
        expect(screen.getByText('Click me')).toBeInTheDocument();
    });

    it('matches snapshot', () => {
        const { container } = render(<SomeComponent />);
        expect(container).toMatchSnapshot();
    });
});