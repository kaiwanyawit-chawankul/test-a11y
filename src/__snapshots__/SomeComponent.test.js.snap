// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SomeComponent matches snapshot 1`] = `
<div>
  <div
    role="main"
  >
    <button>
      Click me
    </button>
  </div>
</div>
`;
